Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Init.hs

```diff
@@ -22,7 +22,7 @@ castVehicleVariant mbVehCategory mbVehVariant = case (mbVehCategory, mbVehVarian
   _ -> Nothing
 
 castPaymentType :: MonadFlow m => Text -> m DMPM.PaymentType
-castPaymentType "ON_FULFILLMENT" = return DMPM.ON_FULFILLMENT
+castPaymentType "ON-FULFILLMENT" = return DMPM.ON_FULFILLMENT
```

Comment:
can we replace `castPaymentType` with `decodeFromText`. We need to have FromJSON instance for this to work